### PR TITLE
API: Isolate the abstract base view controller

### PR DESCRIPTION
**Pull request by ElDeveloper**
This class now lives in its own module `abc-view-controller.js`.

### CODE REVIEW

**Comment by ElDeveloper** [comment]
cc @iprotsyuk

File: emperor/support_files/js/abc-view-controller.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+   *
+   * @class EmperorViewControllerABC
+   *
+   * Initializes an abstract tab. This has to be contained in a DOM object and
+   * will use the full size of that container.  The title represents the title
+   * of the jQuery tab.  The description will be used as help text to describe
+   * the functionality of each subclass tab.
+   *
+   * @param {Node} container Container node to create the controller in.
+   * @param {String} title title of the tab.
```
Comment (antgonza) [inline emperor/support_files/js/abc-view-controller.js]: Just noticed that title and helper are lower case ...

Comment (ElDeveloper) [inline emperor/support_files/js/abc-view-controller.js]: Fixed, thanks!

**Comment by ElDeveloper** [comment]
Friendly bump.

**Comment by mortonjt** [comment]
I think this looks good to go.  @antgonza @wasade @josenavas want to merge this?

**Comment by antgonza** [comment]
Thanks @ElDeveloper and @mortonjt!

**Pull request merged**